htlcswitch/hop/payload.go

```diff
@@ -87,6 +118,18 @@ func NewPayloadFromReader(r io.Reader) (*Payload, error) {
 
 	parsedTypes, err := tlvStream.DecodeWithParsedTypes(r)
 	if err != nil {
+		// Promote any required type failures into ErrInvalidPayload.
+		if e, required := err.(tlv.ErrUnknownRequiredType); required {
+			// NOTE: FinalHop will be incorrect if the unknown
+			// required was type 0. Otherwise, the failure must have
+			// occurred after type 6 and cid should contain an
+			// accurate value.
+			return nil, ErrInvalidPayload{
+				Type:      tlv.Type(e),
+				Violation: RequiredViolation,
+				FinalHop:  cid == 0,
```

Comment:
Is there a use for the `FinalHop` field?

Reply:
for now it is only used to print `final hop` or `intermediate hop` in the failure message